src/Installer/exports/Get-AzModuleWithUpdate.ps1

```diff
@@ -0,0 +1,57 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+function Get-AzModuleWithUpdate {
+    [OutputType([System.Collections.ArrayList])]
+    [CmdletBinding(DefaultParameterSetName = 'Default', PositionalBinding = $false)]
+    param(
+        [Parameter(ParameterSetName = 'Default', Mandatory, HelpMessage = 'The Registered Repostory.')]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        ${Repository}
+    )
+
+    process {
+
+        $modules = New-Object System.Collections.ArrayList
+
+        if ($PSBoundParameters.ContainsKey('Repository')) {
+            $installModules = @{}
+            $null = Get-InstalledModule -Name Az* | ForEach-Object { $installModules[$_.Name] = $_.Version }
```

Comment:
we can do 
$null = Get-InstalledModule -Name Az* | ForEach-Object { $installModules.Add($_) }